sdk/nodejs/src/request.ts

```diff
@@ -43,86 +14,107 @@ const TRACK_PATH = '/v1/track'
 const BUCKETING_URL = `${BUCKETING_BASE}${HOST}`
 const EDGE_DB_QUERY_PARAM = '?enableEdgeDB='
 
+class ResponseError extends Error {
+    status: number
+}
+
+const exponentialBackoff: RequestInitWithRetry['retryDelay'] = (
+    attempt, error, response
+) => {
+    return Math.pow(2, attempt) * 1000
+}
+
+const retryOnRequestError: (retries: number) => RequestInitWithRetry['retryOn'] = (retries) => (
```

Comment:
this double function definition is confusing, any way to split this up to make it more clear what's happening?